common/crypto_utils/avalon_crypto_utils/signature.py

```diff
@@ -365,6 +362,91 @@ def verify_signature(self, input_json, verification_key):
+            order request. Required in 2 step verification.
+        Returns enum type SignatureStatus
+        """
+        if requester_nonce is None:
+            logger.error("Missing requester_nonce argument")
+            return SignatureStatus.FAILED
+
+        decoded_v_key = crypto.base64_to_byte_array(
+                wo_response["extVerificationKey"])
+        v_key = crypto.byte_array_to_string(decoded_v_key)
+        concat_string = v_key + requester_nonce
+        v_key_sig = wo_response["extVerificationKeySignature"]
+        v_key_hash = crypto.compute_message_hash(
+            bytes(concat_string, 'UTF-8'))
+        try:
+            _verifying_key = crypto.SIG_PublicKey(wo_verification_key)
+        except Exception as error:
+            logger.error("Error in verification key of"
+                         "verification key signature : %s", error)
+            return SignatureStatus.INVALID_VERIFICATION_KEY
+        decoded_v_key_sig = crypto.base64_to_byte_array(v_key_sig)
+        sig_result = _verifying_key.VerifySignature(
+            v_key_hash,
+            decoded_v_key_sig)
+        if sig_result == 1:
+            return SignatureStatus.PASSED
+        elif sig_result == 0:
+            return SignatureStatus.FAILED
+        else:
+            return SignatureStatus.INVALID_SIGNATURE_FORMAT
+
+# -----------------------------------------------------------------------------
+    def verify_signature(self, wo_response, wo_res_verification_key,
+                         requester_nonce=None):
+        """
+        Function to verify the signature received from the enclave
+        Parameters:
+            @param wo_response - dictionary contains work order response
+            as per Trusted Compute EEA API 6.1.2 Work Order Result Payload
+            @param wo_res_verification_key - worker ECDSA/SECP256K1
+            public key used to verify work order response signature.
+            @param requester_nonce - requester generated nonce passed in work
+            order request. Required in 2 step verification.
+        Returns enum type SignatureStatus
+        """
+        # if verification_key present in work order response
+        # then do 2 step verification
+        # step1 - The verification key signature from the
+        # response is verified using worker’s public verification key
+        # (aka KME's verification key)
+        if "extVerificationKey" in wo_response:
+            status = self._verify_wo_verification_key_signature(
+                wo_response,
+                wo_res_verification_key,
+                requester_nonce)
+            if status == SignatureStatus.PASSED:
+                # step2 : work order response signature is verified
+                # using the verification key in the response
+                decoded_v_key = crypto.base64_to_byte_array(
+                    wo_response["extVerificationKey"])
+                v_key = crypto.byte_array_to_string(decoded_v_key)
```

Comment:
We can use `extVerificationKey` directly since it's a base64 encoded string.  We don't need `decoded_v_key ` and `v_key` since there is no need to convert 
base64 string -> bytes -> string

Pass `extVerificationKey`  from the response as argument to `_verify_wo_response_signature()`

Reply:
It was needed because verification_key is passed in base64 format..

Reply:
@manju956 could you please confirm which format is used by WPE. AFAIK we need to use public key in base64 string format and append that with requester nonce. https://github.com/hyperledger/avalon/blob/master/common/sgx_workload/workload/ext_work_order_info_kme.cpp#L256